### PR TITLE
Convert `two_dimensional_turbulence.jl` and `internal_wave.jl` literated examples in docs to use CairoMakie

### PR DESCRIPTION
**Pull request by navidcy**
This PR changes two of the literated examples in the docs to use Makie for plotting.

- [x] `two_dimensional_turbulence.jl`
- [x] `internal_wave.jl` 

Also: 
- delete the `geostrophic_adjustment.jl` example since it's not being literated any more
- use `FieldTimeSeries` in `baroclinic_adjustment.jl` example

### CODE REVIEW

**Comment by navidcy** [comment]
As soon as I put `fig[0, :] = Label(fig, title, textsize=24)` then the plot size gets all small and weird... Any ideas?

```Julia
using GLMakie

filename = "internal_wave"

fig = Figure(resolution = (800, 400))

ax = Axis(fig[1, 1];
          xlabel = "x",
          ylabel = "z",
          limits = ((-π, π), (-π, π)),
          aspect = AxisAspect(1))

iter = Observable(0)

using JLD2

file = jldopen(filename * ".jld2")
grid = file["serialized/grid"]

title = @lift(string("ωt = ",
              string(round(file["timeseries/t/" * string($iter)] * ω, digits=3))))

w = @lift(Array(file["timeseries/w/" * string($iter)][:, 1, :]))

x, y, z = nodes((Center, Center, Center), grid)

w_lim = 1e-8
w_levels = range(-w_lim, stop=w_lim, length=10)

contourf!(ax, x, z, w; 
          levels = w_levels,
          colormap = :balance,
          colorrange = (-w_lim, w_lim),
          extendlow = :auto,
          extendhigh = :auto)

fig[0, :] = Label(fig, title, textsize=24)

iterations = parse.(Int, keys(file["timeseries/t"]))

record(fig, filename * ".mp4", iterations, framerate=8) do i
    @info "Plotting iteration $i of $(iterations[end])..."
    iter[] = i
end
```

https://user-images.githubusercontent.com/7112768/168449952-2ce68db5-1e21-4f04-8dc8-a29d2fba6c57.mp4

While with **everything same except** the line `fig[0, :] = Label(fig, title, textsize=24)` I get

https://user-images.githubusercontent.com/7112768/168450064-ab456174-06da-45e4-8674-55f0a3356091.mp4

**Comment by tomchor** [comment]
As someone who doesn't plot using Julia often, I've been hearing a lot about `Makie` being better than `Plots` and other packages. Since this PR is making the change, do you mind briefly explaining the rationale behind this change?

**Comment by tomchor** [comment]
> As soon as I put `fig[0, :] = Label(fig, title, textsize=24)` then the plot size gets all small and weird... Any ideas?

Unfortunately I have no idea. I'd try to change the `textsize` and see what effect that has though. It's the only number you're changing with that line...

File: examples/two_dimensional_turbulence.jl [hunk 1]
```diff
@@ -148,8 +148,8 @@ using JLD2
 file = jldopen(filename * ".jld2")
 grid = file["serialized/grid"]
 
-ω = @lift(Array(file["timeseries/ω/"   * string($iter)][:, :, 1]))
-s = @lift(Array(file["timeseries/s/"   * string($iter)][:, :, 1]))
+ω = @lift(Array(file["timeseries/ω/" * string($iter)][:, :, 1]))
+s = @lift(Array(file["timeseries/s/" * string($iter)][:, :, 1]))
```
Comment (glwagner) [inline examples/two_dimensional_turbulence.jl:152]: Can we use `FieldTimeSeries` instead?

File: examples/internal_wave.jl [hunk 2]
```diff
@@ -140,31 +140,67 @@ run!(simulation)
...
+
+using JLD2
+
+file = jldopen(filename * ".jld2")
+grid = file["serialized/grid"]
+
+title = @lift(string("ωt = ",
+              string(round(file["timeseries/t/" * string($iter)] * ω, digits=3))))
+
+w = @lift(Array(file["timeseries/w/" * string($iter)][:, 1, :]))
```
Comment (glwagner) [inline examples/internal_wave.jl:172]: I think we should use `FieldTimeSeries` here rather than manually constructing things with `JLD2`, etc.

**Comment by glwagner** [comment]
> As someone who doesn't plot using Julia often, I've been hearing a lot about `Makie` being better than `Plots` and other packages. Since this PR is making the change, do you mind briefly explaining the rationale behind this change @navidcy?

We like Makie more. It's unbelievably fast if you have an OpenCL enabled GPU and has lots of support for 3D plotting and even ray tracing.

**Comment by glwagner** [comment]
> As soon as I put `fig[0, :] = Label(fig, title, textsize=24)` then the plot size gets all small and weird... Any ideas?

The index `0` has strange behavior. Try indexing all subaxes in a layout starting from `1`.

**Comment by navidcy** [comment]
> I also think it's probably better to use heatmap for all plots. While contour does look better, heatmap is a lot faster and less finnicky, and we get 95% percent of the educational value of the visualization with a heatmap. I almost never use contour except at the very final stages of visualization for paper plots, so...

OK good! But not for the internal wave. I tried it and you can't really see the phase propagating the other direction.

**Comment by navidcy** [comment]
> I would avoid using index `0` and also using `:` if you can. What happens when you put the label in `fig[1, 1:2]` and the contours in `fig[1, 2]`?

The 0 indexing was not the issue. `tellwidth=false` did the trick!!!

```julia
fig[0, :] = Label(fig, title, textsize=24, tellwidth=false)
```

File: examples/internal_wave.jl (hunk 2 above)
Comment (navidcy) [reply examples/internal_wave.jl]: Done!

File: examples/two_dimensional_turbulence.jl (hunk 1 above)
Comment (navidcy) [reply examples/two_dimensional_turbulence.jl]: Done

**Comment by navidcy** [comment]
> As someone who doesn't plot using Julia often, I've been hearing a lot about `Makie` being better than `Plots` and other packages. Since this PR is making the change, do you mind briefly explaining the rationale behind this change @navidcy?

@tomchor, was the answer by @glwagner what you were after?

**Comment by navidcy** [comment]
The `geostrophic_adjustment.jl` lives in `examples` directory but is not being literated. Should we nuke it? Or add it to the list of examples that are literated by the docs?

**Comment by navidcy** [comment]
I don't have much more time to work on other examples at the moment so let's merge this and open new PR for the rest...

Step by step progress.

**Comment by navidcy** [comment]
https://user-images.githubusercontent.com/7112768/168483174-fa1cf8b6-6b7b-44ad-b287-83efe1650b57.mp4

https://user-images.githubusercontent.com/7112768/168483175-774d4840-da8d-4f42-babe-e60f9db08dc5.mp4

**Review by tomchor — APPROVED** [review]

**Comment by glwagner** [comment]
> The `geostrophic_adjustment.jl` lives in `examples` directory but is not being literated. Should we nuke it? Or add it to the list of examples that are literated by the docs?

We shouldn’t have examples that aren’t built because they can go stale. I suggest deleting it.

**Comment by glwagner** [comment]
Can we also avoid using index 0 even if it works this time?

File: examples/internal_wave.jl [hunk 3]
```diff
@@ -142,6 +142,10 @@ run!(simulation)
 
 w_timeseries = FieldTimeSeries(filename * ".jld2", "w")
 
+# And build the the ``x, y`` grid for plotting purposes.
+
+x, y, z = nodes(w_timeseries)
+
 #-
```
Comment (glwagner) [inline examples/internal_wave.jl:149]: What's this do?

File: examples/internal_wave.jl [hunk 4]
```diff
@@ -161,37 +165,34 @@ nothing #hide
...
+n = Observable(1)
 
 title = @lift(string("ωt = ",
-              string(round(w_timeseries.times[$iter] * ω, digits=2))))
-
-w = @lift(Array(w_timeseries[:, 1, :, $iter]))
+              string(round(w_timeseries.times[$n] * ω, digits=2))))
 
-# We build the coordinates from the saved `grid`.
+w = @lift(interior(w_timeseries[$n], :, 1, :))
```
Comment (glwagner) [inline examples/internal_wave.jl:173]:
```suggestion
w = @lift interior(w_timeseries[$n], :, 1, :)
```

Comment (glwagner) [reply examples/internal_wave.jl:173]: Definitely prefer "paren-less" syntax for macros when possible

File: examples/internal_wave.jl [hunk 5]
```diff
@@ -161,37 +165,34 @@ nothing #hide
...
-           extendlow = :auto,
-           extendhigh = :auto)
+          levels = range(-w_lim, stop=w_lim, length=10),
+          colormap = :balance,
+          colorrange = (-w_lim, w_lim),
+          extendlow = :auto,
+          extendhigh = :auto)
 
-fig[0, :] = Label(fig, title, textsize=24, tellwidth=false)
+fig[1, :] = Label(fig, title, textsize=24, tellwidth=false)
```
Comment (glwagner) [inline examples/internal_wave.jl:187]:
```suggestion
fig[1, 1] = Label(fig, title, textsize=24, tellwidth=false)
```

or is it `1:2`? Is there a colorbar?

File: examples/internal_wave.jl [hunk 6]
```diff
@@ -161,37 +165,34 @@ nothing #hide
...
-iterations = 1:length(w_timeseries.times)
+frames = 1:length(w_timeseries.times)
 
-record(fig, filename * ".mp4", iterations, framerate=8) do i
-    @info "Plotting iteration $i of $(iterations[end])..."
-    iter[] = i
+record(fig, filename * ".mp4", frames, framerate=8) do i
+    @info "Plotting iteration $i of $(frames[end])..."
+    n[] = i
 end
```
Comment (glwagner) [inline examples/internal_wave.jl:196]:
```suggestion
record(fig, filename * ".mp4", frames, framerate=8) do i
       @info "Plotting iteration $i of $(frames[end])..."
       n[] = i
end
```

just aligning

File: examples/two_dimensional_turbulence.jl [hunk 7]
```diff
@@ -133,22 +133,20 @@ axis_kwargs = (xlabel = "x",
...
 
-title = @lift(string("t = ", string(round(times[$iter], digits=2))))
+title = @lift(string("t = ", string(round(times[$n], digits=2))))
 
-# We open the saved output `.jld2` files and extract the vorticity and speed.
-
-ω = @lift(Array(ω_timeseries[:, :, 1, $iter]))
-s = @lift(Array(s_timeseries[:, :, 1, $iter]))
+ω = @lift(interior(ω_timeseries[$n], :, :, 1))
+s = @lift(interior(s_timeseries[$n], :, :, 1))
```
Comment (glwagner) [inline examples/two_dimensional_turbulence.jl:149]:
```suggestion
ω = @lift interior(ω_timeseries[$n], :, :, 1)
s = @lift interior(s_timeseries[$n], :, :, 1)
```

paren-less?

File: examples/two_dimensional_turbulence.jl [hunk 8]
```diff
@@ -162,15 +160,15 @@ s_lim = 0.2
 heatmap!(ax_s, xs, ys, s;
          colormap = :speed, colorrange = (0, s_lim))
 
-fig[0, :] = Label(fig, title, textsize=24, tellwidth=false)
+fig[1, :] = Label(fig, title, textsize=24, tellwidth=false)
 
 # Finally, we record a movie.
 
-iterations = 1:length(times)
+frames = 1:length(times)
```
Comment (glwagner) [inline examples/two_dimensional_turbulence.jl:167]: ❤️

File: examples/two_dimensional_turbulence.jl [hunk 9]
```diff
@@ -162,15 +160,15 @@ s_lim = 0.2
...
 
-record(fig, filename * ".mp4", iterations, framerate=8) do i
-    @info "Plotting iteration $i of $(iterations[end])..."
-    iter[] = i
+record(fig, filename * ".mp4", frames, framerate=8) do i
+    @info "Plotting iteration $i of $(frames[end])..."
+    n[] = i
```
Comment (glwagner) [inline examples/two_dimensional_turbulence.jl:171]:
```suggestion
record(fig, filename * ".mp4", frames, framerate=8) do frame
       @info "Plotting iteration $i of $(frames[end])..."
       n[] = frame
```

now I'm inspired

File: examples/internal_wave.jl (hunk 3 above)
Comment (navidcy) [reply examples/internal_wave.jl:149]: Breaks the cell in literate. Shows the output and then starts new code block.

**Pull request merged**